workflow/workflow_test.go

```diff
@@ -300,6 +301,9 @@ func TestWithDoubleSerializationDetection(t *testing.T) {
 	})
 	assert.NoError(t, err)
 	assert.Equals(t, outputID, "a")
+	// Confirm that, while we did no double-unserializations or double-serializations,
+	// we did do at least one single one.
+	assert.Equals(t, errorDetect.SerializeCnt+errorDetect.UnserializeCnt > 0, true)
```

Comment:
Without assuming anything about the behavior of the engine, it might still be reasonable to assert that the two counters are either equal or differ by exactly 1?

Reply:
i concur with @dbutenhof on this assertion change

Reply:
If it were the case that there were a single data source which was initially serialized, which is then unserialized, possibly reserialized, and then re-unserialized, it might be reasonable to make assertions about the relative values of the two counters (and, even, the sequence of operations).  But, I don't think that that is the case.

Even assuming that the schema is (re)instantiated for each data source (which is probably not a well-founded assumption, as I expect that a single _schema instance_ could be used for multiple fields (to wit, data instances)), there is no reason to think that the engine wouldn't re-unserialize the initial data source (or re-unserialize a re-serialized data more than once).

So, I really don't think the test can do much other than to differentiate between "zero unserializations" and "at least one unserialization" (and the mock schema will detect double-unserializations).

Reply:
I got you, the `+` is like an `or` here for `SerializeCnt > 0 or UnserializeCnt > 0`.

Reply:
I think I'm not entirely convinced by that argument; but while I'm not enthusiastic about the looseness of this assert it's at least infinitesimally better than nothing so I guess I'll just leave this conversation resolved.